2 - RTB/Documentazione interna/Verbali/Esterni/23-11-15/23-11-15.typ

```diff
@@ -0,0 +1,116 @@
+
+=== Differenze tra applicativo desktop e mobile
+Durante la riunione è emerso che vi saranno delle differenze tra la versione desktop e la versione mobile: l'accesso completo alla funzionalità (esempio richiesta di spostamento di prodotti) sarà fornito per la versione desktop, mentre la versione mobile presenterà delle funzioni limitate (magari solo visualizzazione e ricerca). In ogni caso rimane ancora definire nel dettaglio le differenze.
+
+=== Funzionalità desiderate
+Da una prima analisi sono emerse le principali funzionalità desiderate, quali:
+- creazione dell'ambiente 3D del magazzino, articolata in:
+  - determinazione dei parametri dimensionali del magazzino;
+  - determinazione dei bin;
+  - creazione e posizionamento degli scaffali;
+- caricamento dei dati da un database per la popolazione del magazzino;
+- interrogazione del magazzino mediante ricerce per ID, nome, o caratteristiche similari dei prodotti;
+- individuazione visibile (mediante cambio colore o spostamento della camera) di prodotti ricercati;
+- invio di notifiche mediante la chiamata a API per lo spostamento di prodotti nel magazzino.
+
+Durante l'analisi del processo di creazione si è fatta chiarezza relativamente al rapporto tra creazione dell'ambiente 3D e dati presenti nel database:
+sostanzialmente, il primo passaggio è la creazione dell'ambiente 3D. Il database non impone vincoli strutturali, ma bensì è pensato per interfacciarsi con la struttura determinata in fase di creazione. 
+
+=== API
+Per l'invio delle notifiche di richiesta di spostamento, l'applicativo si interfaccerà con delle API. L'implementazione è emerso non essere oggetto del capitolato, e per tale motivazione possiamo porcedere ad una simulazione del comportamento delle stesse con assunzioni riguardo i parametri e le risposte. Durante il meeting, si è individuato un possibile standard. Esempio per lo spostamento di un elemento da un bin ad un altro:
+- Invio all'API delle coordinate di partenza (es: scaffale) e delle coordiante d'arrivo (un altro scaffale) e il prodotto interessato;
+- L'API risponde:
+  - stato 200: contatto con l'API avvenuto con successo: spostamento richiesto con successo;
+  - stato 200: contatto con l'API avvenuto con successo: spostamento non possibile;
+  - risposta ottenuta in formato json.
+- la logica di controllo della possibilità dello spostamento è gestita interamente dalle API.
+\
+== Conferme riguardo lo stack tecnologico individuato;
+Lo stack tecnologico individuato dal team si articola in:
+- MySQL per la gestione del database relazionale;
+- Threejs per l'implementazione della parte grafica (Javascript o typescript);
+- NodeJs per la comunicazione tra applicativo e database;
+- Docker per la conteinerizzazione del prodotto.
+
+Lo stack tecnologico evidenziato rappresenta un punto di partenza, ma può essere soggetto a modifiche. Ad ogni modo, al momento, le tecnologie individuate e proposte hanno trovato l'approvazione del Proponente.
```

Comment:
- ~evidenziato~ discusso
- aggiungi anche che per loro è indifferente usare JS o TS, e anche il discorso su React/Angular/whatever per il frontend

Reply:
Non sapevo bene come integrare il discorso... qualsiasi suggerimento è ben accetto. Intanto ho provato a sintetizzare quanto detto in merito, ossia che ci viene lasciata massima scelta implementativa a riguardo, seppur l'azienda utilizzi typescript e angular.